release/models/platform/openconfig-platform.yang

```diff
@@ -1142,9 +1180,59 @@ module openconfig-platform {
+             /components/component[name=controller1]/controller-card/state/is-powered-off, TRUE
+             /components/component[name=controller1]/state/last-poweroff-reason/trigger, USER_INITIATED
+             /components/component[name=controller1]/state/last-poweroff-reason/details, "User initiated Shutdown"
+             /components/component[name=controller0]/state/last-poweroff-time, 1706590714681765937
+
+             Scenario#2: If controller0 is PRIMARY and controller1 is SECONDARY and if controller0 is set to 
+             config/power-admin-state = POWER_DISABLED by an operator, then controller0 will stay powered-on until the next reboot. 
+             The leaf state/power-admin-state should show as POWER_DIABLED. The boolean flag however, state/is-powered-off 
+             should show as FALSE. The state/last-poweroff-reason/trigger should show as USER_INITIATED. Also the NOS can 
+             update the leaf /state/last-poweroff-reason/details. The leaf state/last-poweroff-time should record the time 
+             when the card powers-off post the next reboot. Following paths
+
+             /components/component[name=controller0]/state/redundant-role, PRIMARY
+             /components/component[name=controller1]/state/redundant-role, SECONDARY
+             /components/component[name=controller0]/controller-card/config/power-admin-state, POWER_DISABLED
+             /components/component[name=controller1]/controller-card/state/power-admin-state, POWER_ENABLED
+             /components/component[name=controller0]/controller-card/state/is-powered-off, FALSE
+             /components/component[name=controller0]/state/last-poweroff-reason/trigger, USER_INITIATED
+             /components/component[name=controller0]/state/last-poweroff-reason/details, "User initiated Shutdown"
+             /components/component[name=controller0]/state/last-poweroff-time, 1706590714681765937
+
+          2. A controller-card which is in state/redundant-role=SECONDARY and is config/power-admin-state=POWER_DISABLED should 
+            remain powered off, even after a reboot. It is possible that after a reboot, both the controller cards are powered-on. 
+            However as soon as the configuration is loaded the NOS should power-off the subject controller-card.
+
+          3. A CONTROLLER_CARD in redundant-role SECONDARY and state/power-admin-state = POWER_DISABLED cannot transition to redundant-role PRIMARY. 
+             If the PRIMARY CONTROLLER_CARD goes down, the device will be offline.
+
+          4. When a controller card boots up and loads it’s configuration, power-admin-state may be set to POWER_DISABLED. The CONTROLLER_CARD should 
+             then power off and never enter into any controller card primary/secondary election process. This also means that an implementation  
+             shouldn't start a controller card election process until the configuration is loaded and consumed.";
```

Comment:
> This also means that an implementation  
             shouldn't start a controller card election process until the configuration is loaded and consumed.";

This assumes/enforces a particular NOS/HW architecture, which isn't valid for all vendors/implementations.

In some implementations, primary/standby selection happens earlier in the boot process, before the full config is loaded (and, often, only a single card is responsible for managing the running-config).

Reply:
can you please ping me? sachendras@google

Reply:
ack, will reach out tomorrow